lib/routes/movie_details.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+import 'package:the_movie_db_flutter/ui_models/movie_preview.dart';
+
+import '../widgets/focus_box.dart';
+import '../widgets/like_button.dart';
+import '../widgets/movie_header.dart';
+import '../widgets/movie_overview.dart';
+import '../widgets/movie_title.dart';
+import '../widgets/vote_average.dart';
+
+class MovieDetails extends StatefulWidget {
+  static const String routeName = "/movie";
+
+  const MovieDetails({super.key});
+
+  @override
+  State<MovieDetails> createState() => _MovieDetailsState();
+}
+
+class _MovieDetailsState extends State<MovieDetails> {
+  static const String genresSeparator = ', ';
+  ColorScheme currentColorScheme = const ColorScheme.light();
+
+  Future<void> _updateImage(ImageProvider provider) async {
+    final ColorScheme newColorScheme =
+        await ColorScheme.fromImageProvider(provider: provider);
+    setState(() {
+      currentColorScheme = newColorScheme;
+    });
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      final MoviePreview movie =
+          ModalRoute.of(context)!.settings.arguments as MoviePreview;
+
+      final posterProvider = NetworkImage(movie.movie.posterUrl);
+      _updateImage(posterProvider);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final MoviePreview payload =
+        ModalRoute.of(context)!.settings.arguments as MoviePreview;
```

Comment:
why is call payload? this basically is a movie, you can get the movie outside build method